lib/src/bar-chart/BarChart.test.tsx

```diff
@@ -0,0 +1,121 @@
+};
+
+describe("Bar Chart component tests", () => {
+  test("onFilterChange is called when the value of the select is changed", async () => {
+    const onFilterChange = jest.fn();
+    const { getByText, getByRole, getAllByRole } = render(
+      <DxcBarChart
+        showFilter
+        onFilterChange={onFilterChange}
+        chartTitle="Bar Chart"
+        series={[
+          {
+            title: "Series 1",
+            type: "bar",
+            data: [
+              { x: 0, y: 100000 },
+              { x: 1, y: 200000 },
+              { x: 2, y: 300000 },
+              { x: 3, y: 400000 },
+              { x: 4, y: 500000 },
+            ],
+          },
+          {
+            title: "Average",
+            type: "threshold",
+            y: 300000,
+          },
+        ]}
+      />
+    );
+    expect(getByText("Bar Chart")).toBeTruthy();
+    expect(getByText("Filter displayed data")).toBeTruthy();
+    await userEvent.click(getByRole("combobox"));
+    expect(getByText("Series 1")).toBeTruthy();
+    expect(getByText("Average")).toBeTruthy();
+    await userEvent.click(getAllByRole("option")[1]);
+    expect(onFilterChange).toHaveBeenCalled();
+    expect(onFilterChange).toHaveBeenCalledWith(["Series 1"]);
+  });
+  test("onHighlightChange is called when a legend's series is hovered/clicked", async () => {
+    const onHighlightChange = jest.fn();
+    const { getByText } = render(
+      <DxcBarChart
+        showLegend
+        legendTitle="Legend"
+        onHighlightChange={onHighlightChange}
+        series={[
+          {
+            title: "Series 1",
+            type: "bar",
+            data: [
+              { x: 0, y: 100000 },
+              { x: 1, y: 200000 },
+              { x: 2, y: 300000 },
+              { x: 3, y: 400000 },
+              { x: 4, y: 500000 },
+            ],
+          },
+          {
+            title: "Average",
+            type: "threshold",
+            y: 300000,
+          },
+        ]}
+      />
+    );
+    expect(getByText("Legend")).toBeTruthy();
+    await userEvent.hover(getByText("Series 1"));
+    expect(onHighlightChange).toHaveBeenCalled();
+    expect(onHighlightChange).toHaveBeenCalledWith("Series 1");
+  });
+  test("onRetry is called when the retry button is clicked", async () => {
+    const onRetry = jest.fn();
+    const { getByText } = render(
+      <DxcBarChart
+        onRetry={onRetry}
+        error="Error"
+        series={[
+          {
+            title: "Series 1",
+            type: "bar",
+            data: [
+              { x: 0, y: 100000 },
+              { x: 1, y: 200000 },
+              { x: 2, y: 300000 },
+              { x: 3, y: 400000 },
+              { x: 4, y: 500000 },
+            ],
+          },
+          {
+            title: "Average",
+            type: "threshold",
+            y: 300000,
+          },
+        ]}
+      />
+    );
+    expect(getByText("Error")).toBeTruthy();
+    await userEvent.click(getByText("Retry"));
```

Comment:
getByRole("button") is higher in priority than getByText

Reply:
done!